### ISSUE TITLE
Progress Tracking

### ISSUE THREAD

**Issue by JackLot** [JackLot/LearnGit.io#19]
I could see two ways of doing this:

1. Automatically store the last visited lesson
2. Surface a checkbox to the user to manually select once they've completed the lesson

The drawback for the first option would be that if the user is just skipping around (i.e. going back and forth between lessons) the last visited lesson might lose it's meaning. Option 2 seems more consistent with learning management system best practices. It would also give users control over if they feel they're done with the lesson.

**Comment by Error-UsernameNotFound** [comment]
Just a thought, but had you considered a mini 'proof of understanding' quizlet that could show if a user has 'passed'/completed that topic/section? I understand that this might be more work than you'd want to commit (no pun intended).

**Comment by JackLot** [comment]
Good idea, I'll create a feature request to track this ([here](https://github.com/JackLot/LearnGit.io/issues/23)).

I'd like to add something like that the end of each module and potentially tie those into a final "certificate of completion" type quiz too.

**Comment by JackLot** [comment]
Proposed design for the lesson page. This checkbox UI would be duplicated below the lesson's written documentation as well as up top.

![lesson page complete UI mock](https://github.com/JackLot/LearnGit.io/assets/1841253/e1225c48-7456-4a33-bfcb-a8d0710be41f)

**Comment by Error-UsernameNotFound** [comment]
If you were looking for feedback – this looks nice and clean.
My only suggestion would be additionally to append the checkmark to the left hand nav bar when the lesson is completed (excuse the rough mock-up).
***@***.***

**Comment by JackLot** [comment]
True. Your mock-up didn't get uploaded to GitHub but I understand what you're getting at. There needs to be a way to surface completion in the left nav, and probably on the homepage and learning track page as well.

**Comment by JackLot** [comment]
Progress is being made on this feature. I have a working prototype and hopefully will be able to release something to you all in the next week or so. Thank you @Error-UsernameNotFound, I have incorporated your checkmarks suggestion into the left nav.

![Screenshot 2024-12-11 at 1 24 31 PM](https://github.com/user-attachments/assets/9870d0e6-35d4-4cb7-93db-c3a62bdd45fc)

**Comment by JackLot** [comment]
This feature is live.

**Comment by JackLot** [comment]
This feature has been live for a few weeks with no reported issues. Marking complete.
